lib/oli/resources/collaboration.ex

```diff
@@ -429,6 +457,92 @@ defmodule Oli.Resources.Collaboration do
     )
   end
 
+  @doc """
+  Returns the list of all posts across a section given a certain criteria.
+
+  ## Examples
+
+      iex> list_posts_in_section_for_instructor("example_section", :need_approval, 0, 5)
+      [%Post{status: :archived}, ...]
+
+      iex> list_posts_in_section_for_instructor("example_section", :need_reponse, 0, 5)
+      []
+  """
+
+  def list_posts_in_section_for_instructor(section_slug, filter, opts \\ [offset: 0, limit: 10])
```

Comment:
I'm not sure what this function would do here, because as far as I can see it has no actions in the body of the function 🤔

Reply:
This function uses pattern matching for different filters, so we need to have a function head that states what the default arguments are. [Here's a link](https://elixir-lang.org/getting-started/modules-and-functions.html#default-arguments) to read more about it 😀